modules/data_release/ajax/FileUpload.php

```diff
@@ -15,11 +15,30 @@
 $DB   = \Database::singleton();
 $user = \User::singleton();
 
-if ($_POST['action'] == 'upload'
-    && $user->hasPermission("data_release_upload")
-) {
+if (!$user->hasPermission('data_release_upload')) {
+    header("HTTP/1.1 403 Forbidden");
+    exit;
+}
+
+if (!file_exists(__DIR__ . "/../user_uploads/")) {
+    error_log(
+        "ERROR: File upload failed. Default user_uploads"
+        . " directory not found."
+    );
+    header("HTTP/1.1 500 Internal Server Error");
+    exit;
+} elseif (!is_writable(__DIR__ . "/../user_uploads/")) {
+    error_log(
+        "File upload failed. Default user_uploads directory"
+        . " does not appear to be writeable."
+    );
+    header("HTTP/1.1 500 Internal Server Error");
+    exit;
+}
+
+if ($_GET['action'] == 'upload') {
     $fileName    = $_FILES["file"]["name"];
-    $version     = $_POST['version'];
+    $version     = $_POST['version'] ? !empty($_POST['version']) : null;
```

Comment:
I don't think this is correct. The value to the right of the `?` will be `true` or `false` (the `bool` return value of `empty()`. `$version` is probably not supposed to be a bool.

Reply:
oooops. I did not put the function empty in the right place...

Reply:
done in the last commit